tests/evaluate/test_evaluation.py

```diff
@@ -699,6 +701,27 @@ def test_dataset_hash(
     assert diabetes_spark_dataset.hash == "ebfb050519e7e5b463bd38b0c8d04243"
 
 
+def test_trace_dataset_hash():
+    # Validates that a dataset containing Traces can be hashed.
+    df = pd.DataFrame(
+        {
+            "request": ["Hello"],
+            "trace": [Trace(info=create_test_trace_info("tr"), data=TraceData([], "", ""))],
+        }
+    )
+    dataset = EvaluationDataset(data=df)
+    assert dataset.hash == "757c14bf38aa42d36b93ccd70b1ea719"
+    # Hash of a dataset with a different column should be different
+    df2 = pd.DataFrame(
+        {
+            "request": ["Hi"],
+            "trace": [Trace(info=create_test_trace_info("tr"), data=TraceData([], "", ""))],
+        }
+    )
+    dataset2 = EvaluationDataset(data=df2)
+    assert dataset2.hash != dataset.hash
```

Comment:
from reading through the code, it looks like the hashes are because their "request" fields are different, and the "trace" fields are skipped.

my guess is something like this would fail, even though they are different traces
```
    df = pd.DataFrame(
        {
            "trace": [Trace(info=create_test_trace_info("tr"), data=TraceData([], "foo", ""))],
        }
    )
    dataset = EvaluationDataset(data=df)
    assert dataset.hash == "757c14bf38aa42d36b93ccd70b1ea719"
    # Hash of a dataset with a different column should be different
    df2 = pd.DataFrame(
        {
            "trace": [Trace(info=create_test_trace_info("tr"), data=TraceData([], "bar", ""))],
        }
    )
    dataset2 = EvaluationDataset(data=df2)
    assert dataset2.hash != dataset.hash
```

in general, we should try to hash all of the items within the dataframe. there may be things that we cannot hash, and i think ur change makes perfect sense for those cases. 

but for things like `Trace` which we maintain, i think we can do better. can we call `to_json()` on to get a string which we can then hash?

Reply:
Agreed, but as discussed offline I'd prefer we do this in a follow-up.